src/app/guest-book/guest-book.service.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+import { GuestBookService } from './guest-book.service';
+import { GuestBookMessage } from './models/guestBookMessage';
+
+describe('GuestBookService', () => {
+  let service: GuestBookService;
+  let httpMock: HttpTestingController;
+
+  const mockMessages: { [key: string]: GuestBookMessage } = {
+    '1': { id: '1', message: 'This is a comment', author: { name: 'John Doe', email: 'john.doe@test.com' } },
+    '2': { id: '2', message: 'This is a comment2', author: { name: 'John Doe2', email: 'john.doe2@test.com' } },
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [GuestBookService]
+    });
+
+    service = TestBed.inject(GuestBookService);
+    httpMock = TestBed.inject(HttpTestingController);
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should fetch guest book messages', () => {
+    service.getGuestBookMessages().subscribe(messages => {
```

Comment:
it's always a good idea to add condition for unsubscribe - e.g. in this case:
 ```suggestion
    service.getGuestBookMessages().pipe(first()).subscribe(messages => {
```